server/src/main/java/org/elasticsearch/script/ScriptContextStats.java

```diff
@@ -43,6 +56,53 @@ public void writeTo(StreamOutput out) throws IOException {
         out.writeVLong(compilations);
         out.writeVLong(cacheEvictions);
         out.writeVLong(compilationLimitTriggered);
+        if (out.getVersion().onOrAfter(Version.V_8_0_0)) {
+            out.writeOptionalWriteable(compilationsHistory);
+            out.writeOptionalWriteable(cacheEvictionsHistory);
+        }
+    }
+
+    public static class TimeSeries implements Writeable, ToXContentFragment {
+        public final long five;
+        public final long fifteen;
+        public final long day;
+
+        public TimeSeries() {
+            this.five = 0;
+            this.fifteen = 0;
+            this.day = 0;
+        }
+
+        public TimeSeries(long five, long fifteen, long day) {
+            this.five = five;
+            this.fifteen = fifteen;
+            this.day = day;
+        }
+
+        public TimeSeries(StreamInput in) throws IOException {
+            five = in.readVLong();
+            fifteen = in.readVLong();
+            day = in.readVLong();
+        }
+
+        @Override
+        public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+            builder.field(Fields.FIVE_MINUTES, five);
+            builder.field(Fields.FIFTEEN_MINUTES, fifteen);
+            builder.field(Fields.TWENTY_FOUR_HOURS, day);
+            return builder;
+        }
+
+        @Override
+        public void writeTo(StreamOutput out) throws IOException {
+            out.writeVLong(five);
+            out.writeVLong(fifteen);
+            out.writeVLong(day);
+        }
+
+        public boolean isEmpty() {
+            return five == 0 && fifteen == 0 && day == 0;
```

Comment:
This could just be `day == 0` right? Since day includes 5m and 15m, if there is anything in either of those, day must be non-zero.

Reply:
Done.